library/system.lisp

```diff
@@ -45,5 +62,142 @@ While the result will always contain microseconds, some implementations may retu
       (cl:sleep n)
       Unit)))
 
+;;;
+;;; Gathering System information
+;;;
+
+(coalton-toplevel
+
+  (define-type SysError
+    "An error message for system errors."
+    (SysError String String))
```

Comment:
Why two strings?

Reply:
Actually since I'm using `Optional` instead of result I just got rid of `SysError` as a whole.